instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/config/Config.java

```diff
@@ -166,4 +171,11 @@ public Properties asJavaProperties() {
   public boolean isAgentDebugEnabled() {
     return getBooleanProperty("otel.javaagent.debug", false);
   }
+
+  public void updateLogCaptureThreshold(String value) {
+    if (value != null && !value.isEmpty()) {
+      getAllProperties().put(LOG_CAPTURE_THRESHOLD, value.toUpperCase());
+      logger.info("Update {} to {}", LOG_CAPTURE_THRESHOLD, value.toUpperCase());
+    }
+  }
```

Comment:
I think this will have a better chance of getting upstreamed:

(config can't do logging upstream because it's used to configure logging, and this way it's not a one-off for a single property)

```suggestion
  public void updateProperty(String key, String value) {
    getAllProperties().put(key, value);
  }
```